resources/qml/QuickSwitcher.qml

```diff
@@ -32,6 +32,18 @@ Popup {
         roomTextInput.forceActiveFocus();
     }
 
+    Shortcut {
+        id: closeShortcut
+
+        sequence: "Ctrl+K"
+        onActivated: {
+            // It seems that QML takes a second or so to clean up destroyed shortcuts, so instead we'll just disable this shortcut
+            // so it doesn't prevent the quick switcher from opening again
+            closeShortcut.enabled = false;
+            quickSwitcher.close();
+        }
+    }
+
```

Comment:
So, something that has annoyed me for a while already is that ctrl-k only works, when you focus the text input. How about we solve both things in one swoop and make ctrl-k global and make it toggle the quick switcher state?

But otoh, you can already close the quick switcher with escape, so I am not sure I like this change, since hitting ctrl-k twice in frustration will now basically do nothing. Why is escape not enough?

Reply:
Generally speaking, a shortcut of this sort should be able to work in reverse. For example:

- The Windows Start menu can be opened and closed using the Windows key.
- KRunner can be opened and closed with the same hotkey.
- Yakuake opens and closes with F12.
- Pressing Alt in most applications will toggle focus on and off of the menubar.

Also, if we make Ctrl+K global, you should not have to press it twice in frustration anymore.

Reply:
Hm, but to me this is more like hitting ctrl-O in an application, which is not reversible afaik. I'm just not sure this adds much benefit, since we already have one universal key to close stuff, Escape, which works to close the room switcher, room directory, emoji and other completers, edits, threads, message search, etc.

Reply:
I'm not sure how opening a menu (which is pretty much what this functions as) is a non-reversible action.